cql3/statements/select_statement.cc

```diff
@@ -1739,6 +1751,10 @@ mutation_fragments_select_statement::do_execute(query_processor& qp, service::qu
 
     auto cl = options.get_consistency();
 
+    validate_for_read(cl);
+
+    validate_aggregate_and_per_partition_limit(*_selection, _per_partition_limit);
```

Comment:
it's a personal preference but for me if all args are object members it probably means this func simply belongs to the class. Especially that there is similar functions already: validate_distinct_selection